### PR TITLE
Improve applist

### PR DESCRIPTION
**Pull request by YoungeunKim**
Make ares-install -F result more readable

:Release Notes:
Make ares-install -F result more readable

:Detailed Notes:
Current result of ares-install -F is hard to read
Show app's detail infomation as ares-device -s option

:Testing Performed:
1. Passed Unit test in OSE & Auto target
2. Passed eslint
3. Verified with CLI commands
  a. Install apps using ares-install
  b. $ ares-install -F
     id : com.domain.app
     transparent : false
    ... 
  c. Check "$ares-install -l" shows correct list
  d. Check "$ares-device -s" on Auto. Check the result format 

:Issues Addressed:
[PLAT-134674] Make ares-install -F result more readable

### CODE REVIEW

**Comment by YoungeunKim** [comment]
code 👍 
verified 👍

File: lib/util/json.js [hunk 1]
```diff
@@ -32,5 +32,32 @@ function readJsonAsync(file) {
         });
 }
 
+function convertJsonToList(orgJson, level){
```
Comment (ssuminahn) [inline lib/util/json.js:35]: how about JsonToList?
Even without "convert", it seems to have enough meaning.
if you agree this, I hope change all of convertJsonToList to JsonToList.

**Review by ssuminahn — COMMENTED** [review]
code :-1: 
verify :+1:

File: lib/util/json.js (hunk 1 above)
Comment (YoungeunKim) [reply lib/util/json.js:35]: @ssuminahn , Good opinion. 
I thought "JsonToList" such as "atoi", too.
The reason is to add "convert"  to "JsonToList" is that other function name are  "readJsonSync, readJsonAsync".
So, I named it like this.

Please consider the reason. Let's have a talk.

File: bin/ares-install.js [hunk 2]
```diff
@@ -149,38 +150,26 @@ function list(){
         let strPkgs = "";
         let cnt = 0;
         if (pkgs instanceof Array) pkgs.forEach(function (pkg) {
-            if (argv.type) {
-                if (argv.type !== pkg.type) {
-                    return;
+            if (!argv.type || (argv.type && (argv.type === pkg.type))) {
```
[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:153]: What error sentencde do you want to print to user?
If the type is not matched, no list is show to user.

File: bin/ares-install.js [hunk 3]
```diff
@@ -149,38 +150,26 @@ function list(){
...
+                    strPkgs += '\n';
                 }
+                strPkgs += pkg.id;
             }
-            if (cnt++ !== 0) strPkgs = strPkgs.concat('\n');
-            strPkgs = strPkgs.concat(pkg.id);
         });
-        console.log(strPkgs);
-        finish(err);
+        finish(err, {msg : strPkgs.trim()});
```
[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:160]: I'll fix them to like : 
```
strPkgs += pkg.id +  '\n';
finish(err, {msg : strPkgs.trim()});
```
okay?

File: bin/ares-install.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:153]: I've already check the argv.type is valid. I cannot see what you mean. 
`!argv.type || (argv.type && (argv.type === pkg.type)`

Let's have a talk.

**Comment by YoungeunKim** [comment]
@ssuminahn , I commented for your review. I'll update the code after having a discussion with you.

File: lib/util/json.js (hunk 1 above)
Comment (YoungeunKim) [reply lib/util/json.js:35]: keep current.

File: bin/ares-install.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ssuminahn) [reply bin/ares-install.js:153]: how about this? (#153~157)

if (argv.type) {
                if(argv.type === pkg.type){
                        strPkgs += pkg.id + '\n';
                }
            }

[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:153]: @ssuminahn , if user do not gives "type", the expected list is that all type of installed apps.
Your code cannot handle the case.

[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:153]: I had a talk about it. Do not need to error handle. Just "not added" the type of app to list.

**Review by ssuminahn — APPROVED** [review]
code :+1: 
verify :+1:

File: bin/ares-install.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:152]: Keep current condition.

File: bin/ares-install.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (YoungeunKim) [reply bin/ares-install.js:156]: Done.

**Comment by YoungeunKim** [comment]
Code :+1: 
Verified :+1:

**Comment by YoungeunKim** [comment]
@ssuminahn , I updated code for printing system error message. 
I also restore the original code to improve readability as your comment. 

Code :+1: 
Verified :+1:

File: bin/ares-install.js [hunk 4]
```diff
@@ -148,10 +148,13 @@ function install() {
 function list(){
     installLib.list(options, function(err, pkgs) {
         let strPkgs = "";
-        if (pkgs instanceof Array) pkgs.forEach(function (pkg) {
-            if (!argv.type || (argv.type && (argv.type === pkg.type))) {
-                strPkgs += pkg.id + '\n';
+        if (Array.isArray(pkgs)) pkgs.forEach(function (pkg) {
```
Comment (ssuminahn) [inline bin/ares-install.js:151]: Please separate if and foreach

if (Array.isArray(pkgs)) {
pkgs.forEach(function (pkg) {

**Review by ssuminahn — COMMENTED** [review]
verify :+1:

**Review by ssuminahn — COMMENTED** [review]
code :+1: 
verify :+1:

File: bin/ares-install.js (hunk 4 above)
Comment (YoungeunKim) [reply bin/ares-install.js]: Done.

**Pull request merged**